flink-tests/src/test/java/org/apache/flink/test/util/TestUtils.java

```diff
@@ -94,6 +102,59 @@ public static void waitUntilJobInitializationFinished(
                 userCodeClassloader);
     }
 
+    public static CheckpointMetadata loadCheckpointMetadata(String savepointPath)
+            throws IOException {
+        CompletedCheckpointStorageLocation location =
+                AbstractFsCheckpointStorageAccess.resolveCheckpointPointer(savepointPath);
+
+        try (DataInputStream stream =
+                new DataInputStream(location.getMetadataHandle().openInputStream())) {
+            return Checkpoints.loadCheckpointMetadata(
+                    stream, Thread.currentThread().getContextClassLoader(), savepointPath);
+        }
+    }
+
+    public static List<Path> findAllSortedExternalizedCheckpoint(File checkpointDir, JobID jobId)
+            throws IOException {
+        return findAllExternalizedCheckpoint(checkpointDir, jobId).stream()
+                .sorted(
+                        Comparator.comparingInt(
+                                path ->
+                                        Integer.parseInt(
+                                                path.getFileName()
+                                                        .toString()
+                                                        .substring(
+                                                                CHECKPOINT_DIR_PREFIX.length()))))
+                .collect(Collectors.toList());
+    }
+
+    public static Optional<Path> findExternalizedCheckpoint(File checkpointDir, JobID jobId)
```

Comment:
Can't some existing methods be used here, e.g. `getMostRecentCompletedCheckpointMaybe`?
Are these methods specific to externalized checkpoints?